File: client/real-estate-crm-react/src/types/unit.ts
export const UnitStatus = {
  Available: "Available",
  Reserved: "Reserved",
  Sold: "Sold",
  Unavailable: "Unavailable",
} as const;
export type UnitStatus = (typeof UnitStatus)[keyof typeof UnitStatus];

export interface Unit {
  id: string;
  projectId: string;
  unitCode: string;
  propertyType: string | null;
  price: number;
  area: number | null;
  bedrooms: number | null;
  bathrooms: number | null;
  floor: string | null;
  location: string | null;
  status: UnitStatus;
  downPayment: number | null;
  installmentYears: number | null;
  description: string | null;
  isPubliclyListed: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateUnitRequest {
  projectId: string;
  unitCode: string;
  propertyType?: string | null;
  price: number;
  area?: number | null;
  bedrooms?: number | null;
  bathrooms?: number | null;
  floor?: string | null;
  location?: string | null;
  status: UnitStatus;
  downPayment?: number | null;
  installmentYears?: number | null;
  description?: string | null;
  isPubliclyListed?: boolean;
}

export type UpdateUnitRequest = CreateUnitRequest;

export interface UnitListQuery {
  page?: number;
  pageSize?: number;
  search?: string;
  status?: UnitStatus;
  projectId?: string;
  propertyType?: string;
  sortBy?: string;
  sortDirection?: "asc" | "desc";
}

export interface UnitImage {
  id: string;
  unitId: string;
  url: string;
  fileName: string;
  contentType: string;
  sizeBytes: number;
  createdAt: string;
}

File: src/RealEstateCRM.Domain/Entities/Unit.cs
using RealEstateCRM.Domain.Common;
using RealEstateCRM.Domain.Enums;

namespace RealEstateCRM.Domain.Entities;

public class Unit : TenantEntity, ISoftDelete
{
    public Guid ProjectId { get; set; }
    public string UnitCode { get; set; } = string.Empty;
    public string? PropertyType { get; set; }
    public decimal Price { get; set; }
    public decimal? Area { get; set; }
    public int? Bedrooms { get; set; }
    public int? Bathrooms { get; set; }
    public string? Floor { get; set; }
    public string? Location { get; set; }
    public UnitStatus Status { get; set; } = UnitStatus.Available;

    /// <summary>Opts this unit into the public, unauthenticated cross-tenant marketplace listing.</summary>
    public bool IsPubliclyListed { get; set; }
    public decimal? DownPayment { get; set; }
    public int? InstallmentYears { get; set; }
    public string? Description { get; set; }

    public bool IsDeleted { get; set; }
    public DateTime? DeletedAt { get; set; }
    public Guid? DeletedBy { get; set; }
}
